benchmarks/bench_methods_run.py

```diff
@@ -248,7 +222,10 @@ def one_run(X, X_shape, random_state, method, n_atoms, n_times_atom, reg):
         ds_init = ds_init[:, 0, :]
 
     # run the selected algorithm with one iter to remove compilation overhead
-    _, _, _, _ = func(X, ds_init, reg, 1, random_state, label)
+    # if dicodile, the workers are started but not stopped or reused, so
+    # doubles the requirement for workers
+    if label != 'dicodile fullrank':
```

Comment:
You will need the same for rank1 dicodile
```suggestion
    if 'dicodile' not in label:
```